src/App.tsx

```diff
@@ -15,36 +16,105 @@
   'Garlic',
 ];
 
+enum SortType {
+  alphabet = 'alphabet',
+  length = 'length',
+  default = '',
+}
+
+interface SortParams {
+  sortBy: SortType;
+  isReversed: boolean;
+}
+
+function getPreparedGoods(goods: string[], { sortBy, isReversed }: SortParams) {
+  const preparedGoods = [...goods];
+
+  if (sortBy !== SortType.default) {
+    preparedGoods.sort((good1, good2) => {
+      switch (sortBy) {
+        case SortType.length:
+          return good1.length - good2.length;
+
+        case SortType.alphabet:
+          return good1.localeCompare(good2);
+
+        default:
+          return 0;
+      }
+    });
+  }
+
+  if (isReversed) {
+    preparedGoods.reverse();
+  }
+
+  return preparedGoods;
+}
+
 export const App: React.FC = () => {
+  const [sortBy, setSortBy] = useState<SortType>(SortType.default);
+  const [isReversed, setIsReversed] = useState(false);
+  const visibleGoods = getPreparedGoods(goodsFromServer, {
+    sortBy,
+    isReversed,
+  });
+
+  const handleResetButton = () => {
+    setSortBy(SortType.default);
+    setIsReversed(false);
+  };
```

Comment:
[CODE STYLE] - The function's name `handleResetButton` should start with a verb that describes its action. A better name would be `handleResetButtonClick` to clearly convey that it handles the click event of the reset button.